rspace/src/test/scala/coop/rchain/rspace/history/RadixTreeSpec.scala

```diff
@@ -0,0 +1,830 @@
+    val notEmptyPart = createBV(s)
+    val emptyPart    = List.fill(32 - notEmptyPart.size.toInt)(0x00.toByte)
+    ByteVector(emptyPart) ++ notEmptyPart
+  }
+
+  def createBV(s: String): ByteVector = ByteVector(Base16.unsafeDecode(s))
+  def createInsertActions(dataSet: List[radixKV]): List[InsertAction] =
+    dataSet.map { ds =>
+      InsertAction(ds.rKey.toSeq, Blake2b256Hash.fromByteVector(ds.rValue))
+    }
+
+  case class radixKV(rKey: ByteVector, rValue: ByteVector)
+
+  object radixKV {
+    def apply(strKey: String, strValue: String): radixKV =
+      new radixKV(createBV(strKey), createBV32(strValue))
+  }
+
+  /*
+        key       |   value
+     111122334455 | 0000...0001
+     11112233AABB | 0000...0002
+     1111AABBCC   | 0000...0003
+     33           | 0000...0004
+     FF0011       | 0000...0005
+     FF012222     | 0000...0006
+   */
+  private val treeDataSet = List(
+    radixKV("111122334455", "01"),
+    radixKV("11112233AABB", "02"),
+    radixKV("1111AABBCC", "03"),
+    radixKV("33", "04"),
+    radixKV("FF0011", "05"),
+    radixKV("FF012222", "06")
+  )
+
+  private val referenceLeafPrefixes = treeDataSet.map(_.rKey)
+  private val referenceLeafValues   = treeDataSet.map(_.rValue)
+
+  private val referenceInsertActions = createInsertActions(treeDataSet)
+
+  private val exportSettings = ExportDataSettings(
+    flagNodePrefixes = true,
+    flagNodeKeys = true,
+    flagNodeValues = true,
+    flagLeafPrefixes = true,
+    flagLeafValues = true
+  )
+  case class ExportParameters(
+      rootHash: ByteVector, // hash
+      typedStore: KeyValueTypedStore[Task, ByteVector, ByteVector],
+      takeSize: Int,     // take size
+      skipSize: Int,     // skip size
+      withSkip: Boolean, // start with skip is true
+      exportData: ExportData,
+      lastPrefix: Option[ByteVector] // last prefix
+  )
+
+  case class MultipageExportResults(
+      firstExport: (ExportData, Option[ByteVector]),
+      reconstructExport: (ExportData, Option[ByteVector])
+  )
+
+  def validateMultipageExport(
+      rootHash: ByteVector,
+      store: KeyValueTypedStore[Task, ByteVector, ByteVector],
+      withSkip: Boolean
+  ): Task[MultipageExportResults] = {
+
+    def multipageExport(
+        parameters: ExportParameters
+    ): Task[Either[ExportParameters, ExportParameters]] = {
+      def collectExportData(prevData: ExportData, pageData: ExportData): ExportData =
+        ExportData(
+          prevData.nodePrefixes ++ pageData.nodePrefixes,
+          prevData.nodeKeys ++ pageData.nodeKeys,
+          prevData.nodeValues ++ pageData.nodeValues,
+          prevData.leafPrefixes ++ pageData.leafPrefixes,
+          prevData.leafValues ++ pageData.leafValues
+        )
+      parameters match {
+        case ExportParameters(
+            rootHash,
+            store,
+            takeSize,
+            skipSize,
+            withSkip,
+            exportData,
+            lastPrefix
+            ) =>
```

Comment:
Instead of this match you can give value directly from this class, e.g.: `if (parameters.withSkip) parameters.skipSize else 0`. For economy you can rename input data `parameters` to e.g `p`: `if (p.withSkip) p.skipSize else 0`. In this case user see that this val it's init data from `p`.